### ISSUE TITLE
What is the difference between max_connections and  max_connection_pools in circuit breaker?

### ISSUE THREAD

**Issue by maxliu007** [envoyproxy/envoy#10634]
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/circuit_breaker.proto#config-cluster-v3-circuitbreakers
About
{
  "priority": "...",
  "**max_connections**": "{...}",
  "max_pending_requests": "{...}",
  "max_requests": "{...}",
  "max_retries": "{...}",
  "retry_budget": "{...}",
  "track_remaining": "...",
  "**max_connection_pools**": "{...}"
}
What is the difference between both? Overlapped?

**Comment by htuch** [comment]
See https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking#arch-overview-circuit-break-cluster-maximum-connection-pools

**Comment by maxliu007** [comment]
In fact, I have read the link you posted before. However I am so confused on when to use the max_connections and max_connection_pools.
1. Is max_connection_pools bound by max_connection_pools? 
2. If I set max_connection_pools=100, while max_connections=300, what will happen?
3. If I set max_connection_pools=300, while max_connections=100, what will happen?
4. Ary they complementary?

With the same terminology, if it is describing a DB connection pool, there will be a parameter to control the connection pool size, e.g. max_db_connection_pool_size. How about for envoy connections? I think that will be max_connection_pools for envoy, am I correct? Then what is the use of max_connections here?

Also, why **max_connection_pools** is in plural instead of **_max_connection_pool_** for a cluster?

**Comment by maxliu007** [comment]
> See https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking#arch-overview-circuit-break-cluster-maximum-connection-pools

In fact, I have read the link you posted before. However I am so confused on when to use the max_connections and max_connection_pools.

Is max_connection_pools bound by max_connection_pools?
If I set max_connection_pools=100, while max_connections=300, what will happen?
If I set max_connection_pools=300, while max_connections=100, what will happen?
Ary they complementary?
With the same terminology, if it is describing a DB connection pool, there will be a parameter to control the connection pool size, e.g. max_db_connection_pool_size. How about for envoy connections? I think that will be max_connection_pools for envoy, am I correct? Then what is the use of max_connections here?

Also, why max_connection_pools is in plural instead of max_connection_pool for a cluster?
